tempesta_fw/addr.c

```diff
@@ -565,3 +518,19 @@ tfw_addr_fmt(const TfwAddr *addr, char *out_buf, size_t buf_size)
 	return (pos - out_buf);
 }
 DEBUG_EXPORT_SYMBOL(tfw_addr_fmt);
+
+ssize_t
+tfw_addr_sa_len(const TfwAddr *addr)
+{
+        sa_family_t family;
+
+        BUG_ON(!addr);
+
+        family = addr->sa.sa_family;
+        if (unlikely(family_is_not_supported(family))) {
```

Comment:
I just had a look at family_is_not_supported() and validate_addr() once more. At which circumstances do we recieve a packet with unsupported family? I believe ip_rcv()/ipv6_rcv(), the earliest point at which Tempesta works, won't get the packet if it has wrong family. So the check is mostly about programming error. So it must be optimized out w/o DEBUG option.

This validate_addr() must use BUG_ON() rather than WARN_ON().

Regarding this code place: there is no need to inspect the field twice, in amily_is_not_supported() and the below.